BuildTools/Tasks/PackageModFile.cs

```diff
@@ -0,0 +1,205 @@
+				continue;
+
+			tmodFile.AddFile($"lib/{dllName}.dll", File.ReadAllBytes(dllPath));
+			modProperties.AddDllReference(dllName);
+		}
+
+		foreach (ITaskItem modReference in modReferences) {
+			string? modName = modReference.GetMetadata("Identity");
+			string? weakRef = modReference.GetMetadata("Weak");
+
+			Log.LogMessage(MessageImportance.Low, $"Adding mod reference with mod name {modName} [Weak: {weakRef}]");
+			modProperties.AddModReference(modName, string.Equals(weakRef, "true", StringComparison.OrdinalIgnoreCase));
+		}
+	}
+
+	private List<ITaskItem> GetNugetReferences() {
+		Dictionary<string, ITaskItem> nugetLookup = PackageReferences.ToDictionary(x => x.ItemSpec);
+		if (nugetLookup.ContainsKey("tModLoader.CodeAssist")) nugetLookup.Remove("tModLoader.CodeAssist");
+
+		List<ITaskItem> nugetReferences = new List<ITaskItem>();
+		foreach (ITaskItem referencePath in ReferencePaths) {
+			string? hintPath = referencePath.GetMetadata("HintPath");
+			string? nugetPackageId = referencePath.GetMetadata("NuGetPackageId");
+			string? nugetPackageVersion = referencePath.GetMetadata("NuGetPackageVersion");
+
+			if (string.IsNullOrEmpty(nugetPackageId)) continue;
+			if (!nugetLookup.ContainsKey(nugetPackageId)) continue;
+
+			Log.LogMessage(MessageImportance.Low, $"{nugetPackageId} - v{nugetPackageVersion} - Found at: {hintPath}");
+			nugetReferences.Add(referencePath);
+		}
+
+		Log.LogMessage(MessageImportance.Normal, $"Found {nugetReferences.Count} nuget references.");
+
+		if (nugetLookup.Count != nugetReferences.Count)
+			Log.LogWarning($"Expected {nugetLookup.Count} nuget references but found {nugetReferences.Count}.");
+		return nugetReferences;
+	}
+
+	private List<ITaskItem> GetModReferences() {
+		List<ITaskItem> modReferences = new List<ITaskItem>();
+		foreach (ITaskItem modReference in ModReferences) {
+			string? modPath = modReference.GetMetadata("HintPath");
+			if (string.IsNullOrEmpty(modPath))
+				modPath = modReference.GetMetadata("ProjectPath");
+			string? modName = modReference.GetMetadata("Identity");
+			string? weakRef = modReference.GetMetadata("Weak");
+			bool isWeak = string.Equals(weakRef, "true", StringComparison.OrdinalIgnoreCase);
+
+			if (string.IsNullOrEmpty(modName))
+				throw new Exception("A mod reference must have an identity (Include=\"ModName\"). It should match the internal name of the mod you are referencing.");
+
+			Log.LogMessage(MessageImportance.Low, $"{modName} [Weak: {isWeak}] - Found at: {modPath}");
+			modReferences.Add(modReference);
+		}
+
+		Log.LogMessage(MessageImportance.Normal, $"Found {modReferences.Count} mod references.");
+		return modReferences;
+	}
+
+	private BuildProperties GetModProperties() {
+		string buildInfoFile = Path.Combine(ProjectDirectory, "build.txt");
+		if (File.Exists(buildInfoFile)) {
+			Log.LogWarning("Using deprecated build.txt file");
+			return BuildProperties.ReadBuildInfo(buildInfoFile);
+		}
```

Comment:
We can even make this backwards compatible 
i.e. modders dropping the build.txt we can just check in-game for these .csproj properties like you're doing in this PR, and use that instead of build.txt 
For those not using .csproj we keep the simple key = value support in build.txt